restfulpy/validation.py

```diff
@@ -0,0 +1,73 @@
+from nanohttp import context
+from nanohttp.exceptions import HttpBadRequest
+
+
+def validate_form(rules):
+    def decorator(func):
+        def func_wrapper(*args, **kwargs):
+
+            def _deny(_fields):
+                """
+                Raise HttpBadRequest if _fields has overlap with inputs.
+                :param _fields:
+                :return:
+                """
+                for field in _fields:
+                    for env in [context.form, context.query_string]:
+                        if env.get(field, None):
+                            raise HttpBadRequest('%s-denied' % field)
+
+            def _exclude(_fields):
+                """
+                Exclude which parameter not in _fields.
+                :param _fields:
+                :return:
+                """
+                for field in _fields:
+                    for env in [context.form, context.query_string]:
+                        if env.get(field, None):
+                            del env[field]
+
+            def _filter(_fields):
+                """
+                Exclude all parameters unless which are in _fields.
+                :param _fields:
+                :return:
+                """
+                for env in [context.form, context.query_string]:
+                    for _key in list(env):
+                        if _key not in _fields:
+                            del env[_key]
+
+            def _only(_fields):
+                """
+                Raise HttpBadRequest if any parameter is in input and not in _fields.
+                :param _fields:
+                :return:
+                """
+                for field in _fields:
+                    if not any(env.get(field, None) for env in [context.form, context.query_string]):
+                        raise HttpBadRequest('%s-denied' % field)
+                for env in [context.form, context.query_string]:
+                    for _input in list(env):
+                        if _input not in _fields:
+                            raise HttpBadRequest('bad-%s' % _input)
+
+            effective_rules = dict(role=[])
+            for rule in rules:
+                if 'role' not in rule or context.identity.is_in_roles(*rule['role']):
+                    for key, parameters in rule.items():
+                        if key not in effective_rules:
+                            effective_rules[key] = []
+                        effective_rules[key].extend(
+                            [param for param in parameters if param not in effective_rules[key]])
+            effective_rules.pop('role')
+
+            for action, fields in effective_rules.items():
+                locals()[''.join(['_', action])](fields)
```

Comment:
nice_trick, but I recommend you to not use the `**kwargs` and define the arguments, with a name to increase the explicitly.